docs/user/spark/sparksql.rst

```diff
@@ -279,3 +281,120 @@ write to a distributed file system like HDFS.
+
+Geometric predicate function calls to Apache Sedona functions can be pushed down to DataStore by GeoMesa SparkSQL:
+
+.. code-block:: scala
+
+    spark.sql("SELECT geom FROM schema WHERE sedona_ST_Intersects(geom, sedona_ST_PolygonFromEnvelope(100.0,20.0,110.0,30.0))").explain()
+    // == Physical Plan ==
+    // *(1) Scan GeoMesaRelation(...,Some([ geom intersects POLYGON ((100 20, 100 30, 110 30, 110 20, 100 20)) ]),None,None) [geom#32] ...
+
+When performing a spatial join, the predicate for joining two datasets should be a function provided by Apache Sedona,
+otherwise Apache Sedona's catalyst optimization rule won't pickup and optimize your join.
+
+.. code-block:: scala
+
+   // This join is accelerated by Apache Sedona as a RangeJoin
+   spark.sql("SELECT linestrings.geom, polygons.the_geom FROM linestrings JOIN polygons ON sedona_ST_Intersects(linestrings.geom, polygons.the_geom)").explain()
+   // == Physical Plan ==
+   // RangeJoin geom#32: linestring, the_geom#101: multipolygon, true
+   // :- *(1) Scan GeoMesaRelation...
+   // +- *(2) Scan GeoMesaRelation...
+
+   // This is just a normal CartesianProduct or BroadcastNestedLoopJoin
+   spark.sql("SELECT linestrings.geom, polygons.the_geom FROM linestrings JOIN polygons ON ST_Intersects(linestrings.geom, polygons.the_geom)").explain()
+   // == Physical Plan ==
+   // CartesianProduct UDF:st_intersects(geom#32, the_geom#101)
+   // :- *(1) Scan GeoMesaRelation...
+   // +- *(2) Scan GeoMesaRelation...
+
+   // Calling DataFrame functions provided by GeoMesa Spark JTS also yields CartesianProduct or BroadcastNestedLoopJoin
+   dfLineString.join(dfPolygon, st_intersects($"geom", $"the_geom")).explain()
+   // == Physical Plan ==
+   // CartesianProduct UDF(geom#32, the_geom#101)
+   // :- *(1) Scan GeoMesaRelation...
+   // +- *(2) Scan GeoMesaRelation...
+
+.. warning::
+
+   ``option("spatial", "true")`` and any other options described in :ref:`spatial_partitioning_and_faster_joins` won't
+   help configuring spatial joins when using Apache Sedona. Please refer to Apache Sedona documentation for available
+   configuration options.
+
+User can also take advantage of Apache Sedona integration when using PySpark, Please make sure that
+
+  - ``apache-sedona`` package was available in your python environment.
+
+  - ``sedona.register`` package was imported AFTER all ``geomesa_pyspark`` packages.
+
+  - ``SedonaRegister.registerAll`` was called AFTER calling ``geomesa_pyspark.init_sql`` or loading DataFrames from GeoMesa DataStore.
+
+Here is an example start-up code for using Apache Sedona integration feature in PySpark:
+
+.. code-block:: python
+
+   import geomesa_pyspark
+   ...
+   from pyspark.sql import SparkSession
+   ...
+   from sedona.register import SedonaRegistrator
+
+   spark = SparkSession.builder.config(...).getOrCreate()
+   ...
+   geomesa_pyspark.init_sql(spark)
+   SedonaRegistrator.registerAll(spark)
```

Comment:
https://github.com/locationtech/geomesa/pull/2802#discussion_r804132557 why is this not registering the functions?

Reply:
https://github.com/locationtech/geomesa/pull/2802/files#diff-8b5907286a459c13b7639108a67cb0aadb6846f1fa79df21cb21cb72173c26b2R91 seems to do it though